### PR TITLE
feat(javascript-sdk): sdks-1715/native-saml-support

### PR DESCRIPTION
**Pull request by ryanbas21**
- feat(redirect): add SML related params to resume method
- test(autoscript-apps,autoscript-suites,mock-api): test-saml-support

# JIRA Ticket

https://bugster.forgerock.org/jira/browse/SDKS-1715

# Description

adds e2e test cases for saml support

# Type of Change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

e2e tests

# Definition of Done

Check all that apply

- [x] Acceptance criteria is met.
- [x] All tasks listed in the user story have been completed.
- [x] Coded to standards.
- [x] Code peer-reviewed.
- [x] Ensure backward compatibility (special attention).
- [x] API reference docs is updated.
- [x] Unit tests are written.
- [x] Integration tests are written.
- [x] e2e tests are written.
- [x] CI build passing on the feature branch.
- [x] Functional spec is written/updated
- [x] contains example code snippets.
- [x] Change log updated.
- [x] Documentation story is created and tracked.
- [x] UI is completed or ticket is created.
- [] Demo to PO and team.
- [x] Tech debts and remaining tasks are tracked in separated ticket(s).

## Documentation

- [ ] Acceptance criteria met
- [ ] Spell-check run
- [ ] Peer reviewed
- [ ] Proofread

### CODE REVIEW

File: packages/javascript-sdk/src/fr-auth/index.ts [hunk 1]
```diff
@@ -108,16 +108,21 @@ abstract class FRAuth {
   ): Promise<FRStep | FRLoginSuccess | FRLoginFailure> {
     const parsedUrl = new URL(url);
```
Comment (cerebrl) [inline packages/javascript-sdk/src/fr-auth/index.ts:109]: I'm starting to think we could really help with grabbing the `authIndexValue` and pass that into our tree config in the options object when we call next.

**Review by cerebrl — CHANGES_REQUESTED** [review]
This looks like a good start, but here are some things I think are missing:

- Where in the server or the test suite are we "testing" if the SDK properly build the request? By that I mean the query parameters as well as the POST body.
- Please add the image of the tree used in the test. I know this is somewhat controversial, but let's stay consistent with what we do until we decide to alter it.
- Let's discuss the idea of also grabbing the `authIndexValue` for calling the right tree
- Can we also test to make sure the persisted state is removed after the resume is called?

File: e2e/autoscript-apps/src/authn-saml/autoscript.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  setTimeout(function () {
+    if (responseKey) {
+      forgerock.Config.set({
+        realmPath,
+        serverConfig: {
+          baseUrl: amUrl,
+          timeout: 5000,
+        },
+        tree,
+      });
```
[the remark this answers is not in the document]
Comment (ryanbas21) [reply e2e/autoscript-apps/src/authn-saml/autoscript.ts:35]: No, it was just from copy pasting the social-login test and using that to build this. Fixing.

File: e2e/mock-api/src/app/routes.auth.js [hunk 3]
```diff
@@ -67,6 +70,8 @@ export default function (app) {
         res.json(selectIdPCallback);
       } else if (req.query.authIndexValue === 'AMSocialLogin') {
         res.json(idpChoiceCallback);
+      } else if (req.query.authIndexValue === 'service') {
+        res.json(authSuccess);
```
[the remark this answers is not in the document]
Comment (ryanbas21) [reply e2e/mock-api/src/app/routes.auth.js:74]: Yeah, i think this is a mistake.

**Review by cerebrl — APPROVED** [review]

**Review by cerebrl — APPROVED** [review]

**Pull request merged**